s3/s3.go

```diff
@@ -43,13 +43,27 @@ type S3ClientOpts struct {
 	Trace           bool
 	RoleARN         string
 	RoleSessionName string
+	UseIRSA         bool
 }
 
 type s3client struct {
 	S3ClientOpts
 	minioClient *minio.Client
 }
 
+// GetWebIdentityCredentials gets web identity credentials
+func GetWebIdentityCredentials(opts S3ClientOpts) (*credentials.Credentials, error) {
+	sess := session.Must(session.NewSession())
+
+	// If env vars are set from ISRA get credentials.
+	creds := stscreds.NewWebIdentityCredentials(sess, os.Getenv("AWS_ROLE_ARN"), "argo-workflows", os.Getenv("AWS_WEB_IDENTITY_TOKEN_FILE"))
```

Comment:
are your about using env vars - this is meant to be a library. could you mandate the are passed in?

Reply:
Mandate as in do a check that they exists before using and then log some message if not? The idea behind IRSA is you setup a something like https://github.com/aws/amazon-eks-pod-identity-webhook which then when k8s creates that pod at admission it injects those env vars.

Reply:
I might be able to find a way to actually just get the session from the aws sdk. The aws sdk actually automatically uses those env vars when you setup a normal s3 service. However because the client used here is actually minio I am not sure 100% if I can get those temp creds without manually looking for the env vars.

Reply:
no, sorry - pass them in as arguments instead of envvars